lib/mongo/protocol/compressed.rb

```diff
@@ -149,6 +157,10 @@ def compress(buffer)
           Zlib::Deflate.deflate(buffer.to_s, @zlib_compression_level).force_encoding(BSON::BINARY)
         elsif @compressor_id == SNAPPY_BYTE
           Snappy.deflate(buffer.to_s).force_encoding(BSON::BINARY)
+        elsif @compressor_id == ZSTD_BYTE
+          # Use compression level of 3 by default as this is what the Java driver does
+          # DRIVERS-600 will allow this to be configurable in the future
```

Comment:
If there is a reason (e.g. published benchmarks) for why 3 should be used, let's link to that. Otherwise let's remove the level and use the default provided by the zstd library.

Reply:
Related: https://github.com/SpringMT/zstd-ruby/issues/28

Reply:
Regarding the default compression level have a look at https://github.com/facebook/zstd/issues/1148, which resulted in https://github.com/facebook/zstd/pull/1174 and https://github.com/facebook/zstd/pull/1174/commits/e34c000e44444b9f8bd62e5af0a355ee186eb21f

```
/*-*************************************
*  Tuning parameters
***************************************/
#ifndef ZSTD_CLEVEL_DEFAULT
#  define ZSTD_CLEVEL_DEFAULT 3
#endif
```

As level 3 is the default in `zstd.h` as well it seems like a sane default (though not the default chosen by `zstd-ruby`)

Reply:
@p if you feel strongly one way or another let me know, otherwise I would suggest leaving the level at 3

Reply:
I've updated the comment in the source regarding why level 3 was chosen

Reply:
If the default level is 3 then I suggest not specifying the level and making use of the default.